src/lib.rs

```diff
@@ -62,21 +66,23 @@ fn parse_reader(reader: &mut dyn Read) -> Result<aggregate_report::feedback, Par
     }
 }
 
-/// This will panic if any of the files are not readable
-/// and will only print an error on the stderr if parsing
-/// of any specific file fails. Use parse for better error handling
+/// This function takes a directory path as an argument. If no dmarc report are found, an empty `Vec` is returned.
+/// Any subdirectories are ignored.
+/// Any files that can not be parsed will be reported stderr
 pub fn parse_dir(path: &Path) -> Vec<aggregate_report::feedback> {
     let mut results = Vec::new();
 
     if path.is_dir() {
         for entry in read_dir(path).unwrap() {
             let entry = entry.unwrap();
             let path = entry.path();
-
-            let result = parse(&path);
-            match result {
-                Ok(result) => results.push(result),
-                Err(error) => eprintln!("could not parse: {:?} because {:?}", path, error),
+            if !path.is_dir() {
```

Comment:
👍

Reply:
There is probably no reason it can't be done that way, I am still quite new to rust. This was just my first initial idea.

Your code below appears to be using the debug form of the strings.

I will do some research and see what I can find.

Reply:
Sounds good. The way you're doing it is fine. I was just wondering if there was a specific reason to strip out the unicode.